### ISSUE TITLE
Logout bug

### ISSUE THREAD

**Issue by hitokiri** [Tivix/django-rest-auth#438]
# code
`class LogoutView(APIView):
    """
    Calls Django logout method and delete the Token object
    assigned to the current User object.

    Accepts/Returns nothing.
    """
    permission_classes = (AllowAny,)

    def get(self, request, *args, **kwargs):
        if getattr(settings, 'ACCOUNT_LOGOUT_ON_GET', False):
            response = self.logout(request)
        else:
            response = self.http_method_not_allowed(request, *args, **kwargs)

        return self.finalize_response(request, response, *args, **kwargs)

    def post(self, request, *args, **kwargs):
        return self.logout(request)

    def logout(self, request):
        try:
            request.user.auth_token.delete()
        except (AttributeError, ObjectDoesNotExist):
            pass

        django_logout(request)

        return Response({"detail": _("Successfully logged out.")},
                        status=status.HTTP_200_OK)`

# Details
Hello, the logout function does not work since it always gives that if I close session but it is not really capturing the data because the user that takes the request is always anonymous.

**Comment by choh3Vuc** [comment]
I don't disagree that logout is bugged. I am having an issue with the logout as well. However, to not be anonymous on logout, you need to logout with your authentication method so logout knows which user you are to log you out.

In my case I am using:

     REST_FRAMEWORK = {
          'DEFAULT_AUTHENTICATION_CLASSES': (
             'rest_framework.authentication.SessionAuthentication',
          ),
         'DEFAULT_PERMISSION_CLASSES': (
             'rest_framework.permissions.IsAuthenticated',
            ),
    }

To login (with curl) I use:

    curl --request POST \
       --cookie "cookies.txt" \
       --cookie-jar "cookies.txt" \
       --url "http://localhost:8000/rest-auth/login/" \
       --header 'content-type: application/json' \
       --data "{\"username\":\"<username>\", \"password\":\"<password>\"}" -vvv

And I get a `{"key": "<key>"}` (with some user info within the payload as well that are not relevant for this topic).
Then using the simliar curl instruction but with GET instead of POST and a different (protected) endpoint, it works:

    curl --request GET \
       --cookie "cookies.txt" \
       --cookie-jar "cookies.txt" \
       --url "http://localhost:8000/api/some_api_endpoint/" \
       --header 'content-type: application/json' \
       --header "Authorization: "Bearer <key>" -vvv

So to actually logout, you need to POST similar to the GET above:

    curl --request POST \
       --cookie "cookies.txt" \
       --cookie-jar "cookies.txt" \
       --url "http://localhost:8000/rest-auth/logout/" \
       --header 'content-type: application/json' \
       --header "Authorization: "Bearer <key>" -vvv

BUT in that case you get the CSRF bug instead (#181 #164 #159) this is still unfixed as far as I can tell. I would also go as far as saying that the login should probably delete the token before creating it but maybe this should be custom. At the very least logout on Session-enable setup should work out of the box.

I would love to offer a pull request for this issue but I'm not that far into this issue at this stage.

In other words, your problem is you should logout with an authenticated user and you don't hence logging out as an anonymous user which has no effect (maybe it's a bug too and it should issue 401 unauthorized? There is no way it could make sense to logout and anonymous user...)

**Comment by choh3Vuc** [comment]
Ok I got it now!

Using the same workflow as my previous comment, while using a cookie, you can find the csrf token in the cookie and now to properly logout you need to add:

    curl --request POST \
       --cookie "cookies.txt" \
       --cookie-jar "cookies.txt" \
       --url "http://localhost:8000/rest-auth/logout/" \
       --header 'content-type: application/json' \
       --header 'X-CSRFToken: <csrf token>' \
       --header "Authorization: "Bearer <key>" -vvv

And it logs out successfully. However, in case the cookie is lost, you would keep the same old token so deleting tokens on login might be a good feature to have. Which maybe should be optionnal from the rest-auth library's perspective.
